internal/handlers/pg/pjson/array.go

```diff
@@ -46,10 +46,20 @@ func (a *arrayType) UnmarshalJSON(data []byte) error {
 		return lazyerrors.Error(err)
 	}
 
+	if len(rawMessages) > 0 && schemas == nil {
```

Comment:
The idea is to have different messages to help understand the root cause of the problem. If such problem happens, the value hasn't been marshaled correctly (and this is why we can't unmarshal it), so this is not a normal/correct error, it's a sign that things go wrong. My idea here is to help simplify debugging when things go wrong.

I believe we don't need a test for it because normally we don't test lazyerrors.

Reply:
After some discussion, I added a test case to cover this. 
The reason for the test case is to check that we don't ignore such errors (which could lead to a wrong interpretation of data).